src/executor/stack/executor.rs

```diff
@@ -264,14 +261,15 @@ impl<'config, 'precompiles, S: StackState<'config>, P: PrecompileSet>
 		}
 	}
 
-	pub fn state(&self) -> &S {
+	pub const fn state(&self) -> &S {
 		&self.state
 	}
 
 	pub fn state_mut(&mut self) -> &mut S {
 		&mut self.state
 	}
 
+	#[allow(clippy::missing_const_for_fn)]
```

Comment:
Why not make this one `const` too?

Reply:
Because I guess, there is a bug in the clippy. In the `into_state` we invoke destructor because `self` is passed by value. After making the `into_state` const there is an error occures with the message `the destructor for this type cannot be evaluated in constant functions`.